.github/workflows/check-issue-has-release.yml

```diff
@@ -0,0 +1,42 @@
+name: Check for Release
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+      - 'feature/**'
+    types:
+      - opened
+      - closed
+      - reopened
+      - synchronize
+  pull_request_review:
+    types: [submitted]
+
+concurrency:
+  group: check-issue-has-release-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-issue-has-release:
+    name: Issue must have a ZenHub release
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    if: false == ( ( github.event_name == 'pull_request' && ( github.event.action == 'closed' || github.event.pull_request.draft == true || contains( github.head_ref, 'dependabot/' ) ) ) || github.event.pull_request.head.repo.fork ) && github.event.review.state == 'approved'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js (.nvmrc)
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+      - name: npm install
+        run: |
+          npm ci
```

Comment:
Arguably the point of the check is in case we _forget_, because we only set this after we've approved the PR, but that's still a fair point.

I'll see about a non-npm-install option 🙂

Reply:
In this case, since we're just using a few packages that can have a stable version, we can cache `node_modules` directly and install just them. By doing this we can run the check in 28 seconds:

<img width="720" alt="CleanShot 2023-09-19 at 00 15 02" src="https://github.com/google/site-kit-wp/assets/90871/e2444f61-d662-4a82-85b4-57e4664b5f08">

I don't think we'll get much faster, so if that isn't fast enough we either need our own action or to nix this issue 😅

Reply:
> I don't think we'll get much faster, so if that isn't fast enough we either need our own action or to nix this issue 😅

I was thinking we can do without the dependencies altogether 😄 I've opened an alternative in https://github.com/google/site-kit-wp/pull/7725

This should be pretty much instant since it only needs to checkout and setup node. LMKWYT!